### ISSUE TITLE
Generate concise Java code for tests

### ISSUE THREAD

**Issue by mximp** [objectionary/eo#1249]
When `+junit` exists in eo code separate mechanism of Java code generation is enabled. It's driven by `junit.xsl` and `rename-junit-inners.xsl`. It generates inner Java classes for each inner EO object which leads to long class names during Java compilation.
The only option is to re-design how we generate Java code for tests.
Probably we need to generate normal (not inner) Java class for each object just like it's done for normal objects.

**Comment by mximp** [comment]
@yegor256 please assisgn

**Comment by yegor256** [comment]
@mximp I think we can stay with inner objects, but do something with their names. Why not?

**Comment by yegor256** [comment]
@Graur please, help with this one

**Comment by mximp** [comment]
> @mximp I think we can stay with inner objects, but do something with their names. Why not?

@yegor256 The issues is that it's Java compiler who is generating the names for inner classes: it prepends inner class name with the outer classes. No matter what names we pick for them with certain level of nesting the compiler will produce long enough `.class` name to fail. 
For example the following is  the result of 40 nesting classes:
```
EOorg.EOeolang.EOexamples.EOtestTest.EOω1blah0.EOω2blah1.EOω3blah2.EOω4blah3.EOω5blah4.EOω6blah5.EOω7blah6.EOω8blah7.EOω9blah8.EOω10blah9.EOω11blah10.EOω12blah11.EOω13blah12.EOω14blah13.EOω15blah14.EOω16blah15.EOω17blah16.EOω18blah17.EOω19blah18.EOω20blah19.EOω21blah20.EOω22blah21.EOω23blah22.EOω24blah23.EOω25blah24.EOω26blah25.EOω27blah26.EOω28blah27.EOω29blah28.EOω30blah29.EOω31blah30.EOω32blah31.EOω33blah32.EOω34blah33.EOω35blah34.EOω36blah35.EOω37blah36.EOω38blah37.EOω39blah38.EOω40blah39
```

**Comment by yegor256** [comment]
@mximp got it. I like the second option: keep all classes in the same file, but reposition them to avoid nesting.

**Comment by yegor256** [comment]
@mximp let's do this?

**Comment by mximp** [comment]
@Graur I believe it was with you.

**Comment by levBagryansky** [comment]
@maxonfjvipon please assign me

**Comment by 0pdd** [comment]
@mximp 2 puzzles [#3266](https://github.com/objectionary/eo/issues/3266), [#3267](https://github.com/objectionary/eo/issues/3267) are still not solved.

**Comment by 0pdd** [comment]
@mximp all 2 puzzles are solved here: [#3266](https://github.com/objectionary/eo/issues/3266), [#3267](https://github.com/objectionary/eo/issues/3267).
